### PR TITLE
Convert README (.md -> .rst)

### LINKED ISSUE #594: README broken on PyPI (must be reStructuredText)
The [package description](https://pypi.python.org/pypi/pip-tools/) on PyPI is unreadable since PyPI expects the README in [reStructuredText](http://www.sphinx-doc.org/en/stable/rest.html) file format and we use MarkDown.

Solution A: Convert to reST
---------------------

1. Rename the current `README.md` to `README.rst`
1. Replace the markdown of the badges and the code samples ([example](https://github.com/Organice/djangocms-maps/blob/master/README.rst))
1. Add a `long_description=read_file('README.rst')` line to `setup.py` ([example](https://github.com/Organice/djangocms-maps/blob/master/setup.py#L50))

Solution B: Process before Upload
-------------------

1. Integrate [pypandoc](https://pypi.python.org/pypi/pypandoc) in `setup.py` ([example](https://github.com/jrief/djangocms-cascade/blob/master/setup.py#L7-L14))
1. Add a `long_description=convert('README.md', 'rst')` line to `setup.py` ([example](https://github.com/jrief/djangocms-cascade/blob/master/setup.py#L49))

------------

Both solutions above will render a nicely formatted, HTML-styled package description on PyPI.

Quality Assurance
--------------

Optionally, you may check your README with [checkdocs](https://github.com/Organice/djangocms-maps/blob/master/tox.ini#L13-L19) before uploading the package to PyPI, because sometimes the reST-to-HTML conversion that PyPI uses fails -- and renders a still hard-to-read, broken, unformatted package description.

### PR DESCRIPTION
**Pull request by bittner**
This PR
- converts the README from MarkDown to reStructuredText 
- adds a check target for the generated HTML (README) to the Tox configuration
- adds `flake8` and `checkdocs` to the Travis build matrix

Fixes #594.

### CODE REVIEW

**Comment by bittner** [comment]
For some reason Travis wouldn't want to run only `flake8` and only `checkdocs` in an included build matrix job. Hence I have split up the matrix build, so that every PIP version now gets its own Travis job.

Though this takes a little bit longer to run now, probably, it enhances readability on Travis. So, I hope the additional change is fine.

**Comment by bittner** [comment]
The failing build is related to #592. This needs to be fixed outside this PR, probably.

I also needs real-world verification whether the relative URL of the illustration in the README will work on PyPI, or whether an absolute URL is needed:

```rest
.. image:: ./img/pip-tools-overview.png
   :alt: pip-tools overview for phase II
```

**Comment by vphilippon** [comment]
Thanks a lot @bittner.

If anyone can deal with #592, that would be great. Otherwise, I'll try to get on it once I can.

I'll also come back for a proper review. At the first glance it seems nice, I'll have to take a closer look at the matrix split, as well as the rendered result.

**Comment by bittner** [comment]
> If anyone can deal with #592, that would be great.

@vphilippon If you could explain which exceptions may occur [on line 110](https://github.com/jazzband/pip-tools/blob/master/piptools/_compat/contextlib.py#L110) the complaint may be easy to fix. Maybe it's not necessary to catch errors, because they indicate something serious has gone wrong.

[The comment](https://github.com/jazzband/pip-tools/blob/master/piptools/_compat/contextlib.py#L97-L100) seems to excuse that there must be a better way, a more readable, self-explanatory implementation to cover the use case. (For example, the `if` clause [on line 94](https://github.com/jazzband/pip-tools/blob/master/piptools/_compat/contextlib.py#L94-L95) can probably be moved into the encapsulated function.)

**Comment by bittner** [comment]
I've replaced the relative image URL by an absolute one that points to the GitHub jazzband account. There seems to be no way to have the image also uploaded to PyPI. So, this change is needed.

File: .travis.yml [hunk 1]
```diff
@@ -5,10 +5,23 @@ python:
   - "3.4"
   - "3.5"
   - "3.6"
-install:
-- travis_retry pip install tox-travis
+  - "pypy"
```
Comment (vphilippon) [inline .travis.yml]: I would like to see `pypy` added in appveyor (for windows) too, in order to keep the tests the same.

**Comment by vphilippon** [comment]
@bittner I think your rebase overwrote the changes from PR #589, (sorry about that mess 😞 ).
You'll have to put those back in the README.rst.

I think that if you click the "allow edit to maintainers", I could do it for you, as I made the choice to merge the other PR first. Tell me if you want me to take care of it.

**Comment by bittner** [comment]
The PR should be alright now. Ready for merging.

**Comment by vphilippon** [comment]
It seems `pypy` is not available on the windows builder. You can just go ahead and remove it for now, I'll merge later tonight once it get green.

(I sure would like Jazzband to give its members access to kill build jobs right now).

**Comment by bittner** [comment]
One day someone may want to add [some PyPy install commands](https://stackoverflow.com/questions/30822873/how-do-i-install-pypy-on-appveyor) to the AppVeyor configuration.

**Comment by vphilippon** [comment]
Thank you @bittner 👍

**Comment by bittner** [comment]
You're welcome!

The world becomes a better place when you do a good thing from time to time. :smiley:

**Comment by bittner** [comment]
Looks like you have to re-trigger the build. Travis has hiccups.

**Comment by vphilippon** [comment]
Thanks you for the heads-up

**Pull request merged**